doc/en/reference.rst

```diff
@@ -1192,6 +1192,45 @@ passed multiple times. The expected format is ``name=value``. For example::
         [pytest]
         junit_suite_name = my_suite
 
+.. confval:: log_auto_indent
+
+    Allow selective auto-indentation of multiline log messages.
+
+    Supports command line option ``--log-auto-indent [value]``
+    and config option ``log_auto_indent = [value]`` to set the
+    auto-indentation behavior for all logging.
+
+    ``[value]`` can be:
+        * True or On - Dynamically auto-indent multiline log messages
+        * False or Off or 0 - Do not auto-indent multiline log messages
+        * [positive integer] - auto-indent multiline log messages by [value] spaces
+
+    .. code-block:: ini
+
+        [pytest]
+        log_auto_indent = False
+
+    Supports passing kwarg ``extra={"auto_indent": [value]}`` to
+    calls to ``logging.log()`` to specify auto-indentation behavior for
+    a specific entry in the log. ``extra`` kwarg overrides the value specified
+    on the command line or in the config.
+
+    Some examples of using the ``extra`` kwarg might be:
+
+    .. code-block:: python
+
+        import logging
+
+        log = logging.getLogger(__name__)
+
+
+        def test_multiline():
+            log.debug("line 1\nline 2 indented", extra={"auto_indent": True})
+            log.warning("line 1\nline 2 not indented", extra={"auto_indent": "off"})
+            log.log(
+                logging.ERROR, "line 1\nline 2 indented by 5 spaces", extra={"auto_indent": 5}
+            )
```

Comment:
I'm not sure if we need an usage example for the `extra` kwarg. I think the paragraph starting at L1213 explains the feature pretty well and doesn't need an additional code block.

Reply:
done